src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        ShopStorage storage = new ShopStorage();
```

Comment:
Replace `new ShopStorage()` with `ShopStorage.getInstance()` to ensure that the storage is used as a singleton throughout the application. This is crucial for maintaining consistent fruit quantities across operations.